### PR TITLE
Nibe F750 parsing ++

### PR DESCRIPTION
**Pull request by bakrus**
I'm not very skilled in Java programming, but I have managed to get it running with my Nibe F750. Hopefully someone will easily arrange NibeHeatPumpBinding.java to actually utilize both the original VARIABLE_INFO_F1145_F1245-parsing and my new VARIABLE_INFO_F750-parsing. Maybe a new configuration parameter to select which one you have?

The next step could be to make NibeHeatPumpDataParser.java parse a CSV-files from NIBE ModbusManager (Nibe F750 variable information.csv) on startup to make the binding fully adaptable to any Nibe heatpump.

### CODE REVIEW

File: bundles/binding/org.openhab.binding.nibeheatpump/src/main/resources/Nibe F750 variable information.csv [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+"SAM supply air curve: outdoor temp T2";"The supply air curve is defined by 3 supply air temperatures at 3 different outdoor temperatures T1, T2 and T3.";48159;"�C";s16;10;-390;190;0;R/W;
+"SAM supply air curve: outdoor temp T1";"The supply air curve is defined by 3 supply air temperatures at 3 different outdoor temperatures T1, T2 and T3.";48160;"�C";s16;10;-400;200;-150;R/W;
+"SAM supply air curve: supply air temp at T3";"The supply air curve is defined by 3 supply air temperatures at 3 different outdoor temperatures T1, T2 and T3.";48161;"�C";s16;10;160;520;220;R/W;
+"SAM supply air curve: supply air temp at T2";"The supply air curve is defined by 3 supply air temperatures at 3 different outdoor temperatures T1, T2 and T3.";48162;"�C";s16;10;160;520;220;R/W;
+"SAM supply air curve: supply air temp at T1";"The supply air curve is defined by 3 supply air temperatures at 3 different outdoor temperatures T1, T2 and T3.";48163;"�C";s16;10;160;520;220;R/W;
+"SCA accessory";"Activates the SCA accessory";48201;"";u8;1;0;1;0;R/W;
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.nibeheatpump/src/main/resources/Nibe F750 variable information.csv]: this file seem to have some special character issues … "°C" seem to be checked in correctly

File: distribution/openhabhome/configurations/openhab_default.cfg [hunk 2]
```diff
@@ -782,29 +782,26 @@ ntp:hostname=ptbtime1.ptb.de
...
+#modbus:tcp.slave1.length=
 
 # Value type, required for combined registers (details: http://www.simplymodbus.ca/FAQ.htm#Types)
 # Can be "bit", "int8", "uint8", "int16", "uint16", "int32", "uint32", "float32"
 # (optional, defaults to 'uint16')
-#modbus:slave1.valuetype=
+#modbus:tcp.slave1.valuetype=
 
 ############################### PLC Bus Binding #######################################
 #
```
Comment (teichsta) [inline distribution/openhabhome/configurations/openhab_default.cfg]: this changes doesn't seem to be related to NIBE so please create another PR for them …

File: bundles/binding/org.openhab.binding.tcp/src/main/java/org/openhab/binding/tcp/protocol/ProtocolBindingProvider.java [hunk 3]
```diff
@@ -40,7 +40,7 @@
 	/* Get a list of the openHAB Commands associated with the combination of the given Item and protocol command */
 	public List<String> getItemNames(String protocolCommand);
 
-	/* Get a list of the openHAB Commands associated with the combingation of the given Item and protocol command */
+	/* Get a list of the openHAB Commands associated with the combination of the given Item and protocol command */
```
Comment (teichsta) [inline bundles/binding/org.openhab.binding.tcp/src/main/java/org/openhab/binding/tcp/protocol/ProtocolBindingProvider.java]: this changes doesn't seem to be related to NIBE so please create another PR for them …

**Comment by teichsta** [comment]
Hi @bakrus thanks for this contribution. Besides the fact that this PR can't be merged now (since it isn't completed now) if send some review comments. @paulianttila can help finalising this enhancement? Thanks, Thomas E.-E.

File: distribution/openhabhome/configurations/openhab_default.cfg (hunk 2 above)
Comment (bakrus) [inline distribution/openhabhome/configurations/openhab_default.cfg]: I'm new to GitHub and actually not sure how to separate these changes into separate PRs. Hopefully I won't need to make a separate fork for each small change just because it's about different modules?

Comment (teichsta) [inline distribution/openhabhome/configurations/openhab_default.cfg]: no, you can do all changes in the same fork but it is good practice to separate them into different branches and send a PR from the specific branch. Otherwise it would be fairly hard to revert certain particular changes when needed.

File: bundles/binding/org.openhab.binding.nibeheatpump/src/main/resources/Nibe F750 variable information.csv (hunk 1 above)
Comment (teichsta) [inline bundles/binding/org.openhab.binding.nibeheatpump/src/main/resources/Nibe F750 variable information.csv]: this file seem to have to wrong encoding. Please make sure all characters are displayed properly

**Comment by teichsta** [comment]
@paulianttila any chance to have a look into these changes?
@bakrus once all changes have been made to this PR please finally squash your commits into one commit

Thanks, Thomas E.-E.

**Comment by paulianttila** [comment]
I try to look the changes during this week

**Comment by bakrus** [comment]
@teichsta Now I kindof think both the Nibe F750 definition and the 32bit variables are working. Probably @paulianttila should take a look on the changes.

Finally I guess I will need to find out how to squash all my small commits into one... Probably it should be made into two commits. One for the F750 definition and one for the 32bit variables, but it might be challenging for me to separate these things since I have been working in parallell on both.

**Comment by teichsta** [comment]
> Finally I guess I will need to find out how to squash all my small commits into one

yes, please do :-)

> Probably it should be made into two commits

i would be fine with that, too

so let's first see what @paulianttila's brings up.

Thanks, Thomas E.-E.
